### ISSUE TITLE
compatibleType() might be clearer than inhabitsType()

### ISSUE THREAD

**Issue by dalnefre** [tristanls/ex#3]
https://github.com/tristanls/ex/blob/810dabcc24c6ac397176d7be9374ab09f4382d48/ex.js#L80

...or maybe compatibleWith() or compatibleWithType()

**Comment by dalnefre** [comment]
the reverse case, from the perspective of the type, could be Type.hasInstance() or Type.compatibleWith()

it would be nice if one could be derived from the other, for consistency.

**Comment by dalnefre** [comment]
It seems like your notion of "type" is based on compatibility, that is, where can I use this instance based on what methods it supports. As opposed to some kind of group/set membership concept.

**Comment by tristanls** [comment]
Yes. I guess I'm channelling ["dynamics trumps semantics" world view](https://medium.com/@tristanls/dynamics-trumps-semantics-9e7368756ee2).

**Comment by dalnefre** [comment]
I wonder how this affects "types" like, finite, unbounded, terminating, non-terminating, undecidable, side-effect, may cause side-effects, may NOT cause side-effects.

**Comment by tristanls** [comment]
![mind-blown](https://user-images.githubusercontent.com/730640/38113731-71b13fec-336b-11e8-819b-bec23edd390a.gif)
